Frends.AmazonS3.UploadObject/Frends.AmazonS3.UploadObject/Definitions/Options.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+
+namespace Frends.AmazonS3.UploadObject.Definitions
+{
+    /// <summary>
+    /// Upload options.
+    /// </summary>
+    
+    public class Options
+    {
+        /// <summary>
+        /// Filter options by authentication method.
+        /// </summary>
+        /// <example>AWS Credentials</example>
+        [DefaultValue(AuthenticationMethod.AWSCredentials)]
+        public OptionsFor OptionsFor { get; set; }
```

Comment:
If options depend on the authentication method selected in other class, they should be in the same class. Don't worry about making the class big.